docs/user_guide/customizing_dependencies/imagespec.md

```diff
@@ -103,3 +103,9 @@ or
 image = ImageSpec(registry="ghcr.io/flyteorg", packages=["pandas"]).force_push()
 ```
 [flytesnacks]: https://github.com/flyteorg/flytesnacks/tree/master/examples/customizing_dependencies/
+
+## Getting source files into ImageSpec
+Typically, getting source code files into a task's image at run time on a live Flyte backend is done through the fast registration mechanism.
+If the `source_root` and `copy` fields to an `ImageSpec` are left blank, then whether or not your source files are copied into the built `ImageSpec` image is basically the inverse of whether fast register is used. Please see [registering workflows](https://docs.flyte.org/en/latest/flyte_fundamentals/registering_workflows.html#containerizing-your-project) for the full explanation.
+
+Keep in mind that because files are sometimes copied into the built image, the tag that is published for an ImageSpec will change based on whether fast register is enabled, and the contents of any files copied.
```

Comment:
```suggestion
Since files are sometimes copied into the built image, the tag that is published for an ImageSpec will change based on whether fast register is enabled, and the contents of any files copied.
```